### DISCUSSION TITLE
다른 플레이어에게 오브젝트를 보이게 하는 방법?

### DISCUSSION THREAD

**Discussion by hwisulee** [naverz/zepeto-studio-kor#1102] [5. Q&A]
플레이어가 버튼과의 상호작용을 통해 복제된 물체를 손에 들게 하고 싶습니다.
클라이언트 단에선 GetBoneTransform과 GameObject.Instantiate을 통해 플레이어의 손에 물체를 붙이는 것은 성공했습니다.
그러나 서버를 통해 다른 플레이어에게는 보이지 않는다는 것이 문제네요.

이전 작업에서 각 플레이어가 서로의 제스처를 보이게 하는 것은 구현했기에 어떻게 서버에 send하고 받는지는 알고 있습니다.
다만, 서버 단에 "물체에 대한 정보"를 어떻게 send 시켜서 다른 플레이어에게도 보이게 해야할지 감이 오질 않는군요.

지금은 물체를 다음 사진과 같이 send하고 있습니다.

![image](https://user-images.githubusercontent.com/62528282/182052299-278c8b0b-5ccc-4e5a-8d7c-a416deab2e66.png)

잘 받아오는지 테스트를 위해 MessageHandler에 console.log를 사용하여 해당 물체를 받아올 경우 해당 물체의 이름을 받아오게 해놨습니다.

![image](https://user-images.githubusercontent.com/62528282/182052777-086cd26f-789f-4e05-9844-9c4b8857ea0b.png)

다음과 같은 에러가 뜨네요. 

![image](https://user-images.githubusercontent.com/62528282/182052759-3a6d46c3-4df5-4062-baa4-bc9799e9277c.png)

저런 식으로 물체를 보내서 받아오는 건 안되는 건가요?

---

요약 : 상호작용을 통해 물체를 플레이어 손에 들게 하고 싶음.

문제 1. 클라이언트 단에선 보이게 하는 것 성공. 다만 다른 플레이어에겐 안보임. (서버 단에 잘못보내는 것 같음.)
문제 2. 그러나 서버 단에 어떻게 GameObject를 보내야할지 모르겠음. (transform이나 number, string은 보내봤어도..)

**Comment by whild** [comment]
복제하고싶은 물건을 Client코드에 전부 넣어두시고 => 
들고있는 오브젝트 네임을 index를 통해서 다른 클라이언트로 보내고 => 
다른 클라이언트에서 Client코드에서 든 오브젝트와 플레이어(캐릭터)를 찾은 뒤 =>
그 캐릭터에 따로 생성시켜주면 보일 것 같습니다.

**Reply by hwisulee** [reply]
고맙습니다. 덕분에 실마리를 얻어 해결했습니다.
다른 오류가 뜨긴 하는데 일단 상대방에게도 물체가 보이긴 하며 들고 돌아다닐 수 있습니다.
